substrate/frame/contracts/src/storage.rs

```diff
@@ -274,14 +274,15 @@ impl<T: Config> ContractInfo<T> {
 
 	/// Calculates the weight that is necessary to remove one key from the trie and how many
 	/// of those keys can be deleted from the deletion queue given the supplied weight limit.
-	pub fn deletion_budget(weight_limit: Weight) -> (Weight, u32) {
+	pub fn deletion_budget(meter: &WeightMeter) -> (Weight, u32) {
```

Comment:
I feel there is an opportunity to simplify this whole code by eradicating this function. Can we replace this by just incrementally consuming weight from the meter? Maybe this function will just be `fn weight_per_key` if this is needed in different places.

Reply:
We kind of need a budget here so that we can specify up to how many keys we can clear with kill_storage